src/contexts/CheckContext.js

```diff
@@ -14,45 +14,85 @@ export const CheckProvider = ({ children }) => {
+        try {
+            if (checkServerVersion(check)) {
+                let value = await check.checker.apply(context)
+                value.timestamp = new Date()
+                updatedMap.set(check.id, [check, value])
+                updatedStatusResult[value['status']] += 1
+            } else {
+                updatedStatusResult['unavailable']++
+                updatedMap.set(check['id'], [check, { status: 'unavailable', message: `ZMT cannot run this check, because your server doesn't meet version requirements in this checkfile. Min: ${check.min_server_version}, Max: ${check.max_server_version}`, timestamp: new Date() }])
+            }
+        } catch (e) {
+            console.log(e)
+            updatedStatusResult['error']++
+            updatedMap.set(check['id'], [check, { status: 'error', timestamp: new Date() }])
+        }
         setCheckResult(updatedMap)
         setStatusResult(updatedStatusResult)
     }
 
     const runAllCheck = () => {
+        // before running all checks, reset the timer
+        if (checkTimeout) clearTimeout(checkTimeout)
         setIsChecking(true)
         let newMap = new Map()
-        let newStatusResult = { 'error': 0, 'warning': 0, 'healthy': 0, 'inactive': 0 }
+        let newStatusResult = { 'error': 0, 'warning': 0, 'healthy': 0, 'inactive': 0, 'unavailable': 0 }
         let newCheckObject = {}
+        // iterate through all checks
         checks.forEach((check, index) => {
             (async function () {
                 check.id = `zmt-${index}`
                 newCheckObject[check['id']] = check
-                // catch error if check function is not working correctly
+                // if active, run the check
                 if (!inactiveChecks.has(`zmt-${index}`)) {
                     try {
-                        let value = await check.checker.apply(context)
-                        value.timestamp = new Date()
-                        newMap.set(check['id'], [check, value])
-                        newStatusResult[value.status]++
+                        // enforce minimum/maximum server version
+                        if (checkServerVersion(check)) {
+                            let value = await check.checker.apply(context)
+                            value.timestamp = new Date()
+                            newMap.set(check['id'], [check, value])
+                            newStatusResult[value.status]++
+                        } else {
+                            // TODO: Shall we tell admin which server version doesn't fall into the range?
+                            newStatusResult['unavailable']++
+                            newMap.set(check['id'], [check, { status: 'unavailable', message: `ZMT cannot run this check, because your server doesn't meet version requirements in this checkfile. Min: ${check.min_server_version}, Max: ${check.max_server_version}`, timestamp: new Date() }])
```

Comment:
let's try...
`This check is unavailable because the iRODS Server version (4.2.10) is out of range (4.2.0 - 4.2.5)`